src/core/mac/sub_mac.cpp

```diff
@@ -730,5 +794,130 @@ const char *SubMac::StateToString(State aState)
 
 // LCOV_EXCL_STOP
 
+//---------------------------------------------------------------------------------------------------------------------
+// CSL Receiver methods
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+void SubMac::SetCslChannel(uint8_t aChannel)
+{
+    mCslChannel = aChannel;
+}
+
+void SubMac::SetCslPeriod(uint16_t aPeriod)
+{
+    mCslPeriod = aPeriod;
+}
+
+void SubMac::SetCslTimeout(uint32_t aTimeout)
+{
+    mCslTimeout = aTimeout;
+}
+
+void SubMac::FillCsl(Frame &aFrame)
+{
+    uint8_t *cur = aFrame.GetHeaderIe(Frame::kHeaderIeCsl);
+
+    if (cur != NULL)
+    {
+        CslIe *csl = reinterpret_cast<CslIe *>(cur + sizeof(HeaderIe));
+
+        csl->SetPeriod(mCslPeriod);
+        csl->SetPhase(GetCslPhase());
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        otLogDebgMac("%10lu:%s seq=%u phase=%hu", static_cast<uint32_t>(otPlatTimeGet()), __func__,
+                     aFrame.GetSequence(), csl->GetPhase());
+#endif
+    }
+}
+
+void SubMac::HandleCslTimer(Timer &aTimer)
+{
+    aTimer.GetOwner<SubMac>().HandleCslTimer();
+}
+
+void SubMac::HandleCslTimer(void)
+{
+    switch (mCslState)
+    {
+    case kCslSample:
+    {
+        mCslState = kCslSleep;
+        IgnoreError(Sleep());
+        // OT_US_PER_TEN_SYMBOLS: computing CSL Phase using floor division.
+        mCslTimer.StartAt(mCslSampleTime, mCslPeriod * OT_US_PER_TEN_SYMBOLS - OT_US_PER_TEN_SYMBOLS);
+        otLogDebgMac("CSL sleep %u", mCslTimer.GetNow().GetValue());
+        break;
+    }
+
+    case kCslSleep:
+    {
+        mCslState = kCslSample;
+
+        mCslSampleTime += mCslPeriod * OT_US_PER_TEN_SYMBOLS;
+        otPlatRadioUpdateCslSampleTime(mCslSampleTime.GetValue());
```

Comment:
All `otPlatRadio` APIs are exposed to OpenThread core module through `Radio` class in `radio.hpp`. 
- This allows us to easily debug (add logs in these method) 
 - Enable new feature (e.g. OTNS uses some these method to track radio activity/commands) 
- This is also important/useful for multi-radio support.

I highly suggest following the same model and adding all these new CSL related function in `Radio` class. Basically `Mac`/`SubMac` or other core module should not directly call any of `otPlatRadio{Function}()` functions.